ui/react-components/components/AppointmentEditor/AppointmentEditor.jsx

```diff
@@ -348,20 +363,41 @@ const AppointmentEditor = props => {
 
                         </div>
                         <div data-testid="recurrence-type-group">
+                            <div>
                             <div className={classNames(dateHeading)}>
                                 <Label translationKey="REPEATS_EVERY_LABEL" defaultValue="Repeats Every"/>
                             </div>
                             <div>
-                                <RecurrenceTypeRadioGroup
-                                    onChange={event => setRecurrenceType(event.currentTarget.value)}
-                                    onPeriodChange={value => {
-                                        setPeriod(value);
-                                        setRecurrencePeriodError(false);
-                                    }}
-                                    period={period}
-                                    recurrenceType={recurrenceType}/>
-                                <ErrorMessage
-                                    message={recurrencePeriodError ? recurrencePeriodErrorMessage : undefined}/>
+                                <div>
+                                    <RecurrenceTypeRadioGroup
+                                        onChange={event => {
+                                            setRecurrenceType(event.currentTarget.value);
+                                            setWeekDaysError(weekDaysError && event.currentTarget.value === 'WEEK');
+                                        }}
+                                        onPeriodChange={value => {
+                                            setPeriod(value);
+                                            setRecurrencePeriodError(false);
+                                        }}
+                                        period={period}
+                                        recurrenceType={recurrenceType}/>
+                                    <ErrorMessage
+                                        message={recurrencePeriodError ? recurrencePeriodErrorMessage : undefined}/>
+                                </div>
+                                <div className={classNames(weekDaysContainer)}>
+                                    <ButtonGroup buttonsList={weekDays} onClick={buttonKey => {
```

Comment:
Try to extract onClick stuff, into one method. So much is happening is there.

Reply:
All that is to create a new state and set state